src/mixxx.cpp

```diff
@@ -1510,6 +1515,27 @@ void MixxxMainWindow::initActions()
     }
 }
 
+void MixxxMainWindow::slotUpdateWindowTitle(TrackPointer pTrack)
+{
+    QString title;
+    if (pTrack) {
+        QString trackTitle = pTrack->getTitle();
+        QString trackArtist = pTrack->getArtist();
+
+        //Displaying title as "Track Title" by "Artist" - "Current Version"
+        //Or "Track Title - "Current Version" in case the artist field is empty
+        if (trackArtist.size() != 0) {
+            title = trackArtist + " - " + trackTitle;
+        } else {
+            title = trackTitle;
+        }
+        title = title + " - " + m_MixxxVersion;
```

Comment:
This is probably the Reason Banshee uses "by". This also fixes the other issue that you see only the first snipped in the task bar, the track title is more significant than the artist.